main.c

```diff
@@ -48,20 +47,61 @@ char type_to_char(enum tile_type type);
 int main(void) {
 
     printf("Welcome to Frogger Game!\n");
+
     struct board_tile game_board[SIZE][SIZE];
 
-    // TODO (Stage 1.1) Initialise the gameboard.
-    
-    // Read user input and place turtles.
+    // Initialize the game board
+    initialize_board(game_board);
+
+    // Read user input and place turtles
+    int num_turtles;
     printf("How many turtles? ");
-    // TODO (Stage 1.2): Scan in the turtles, and place them on the map.
+    scanf("%d", &num_turtles);
+    place_turtles(game_board, num_turtles);
 
-    // Start the game and print out the gameboard.
+    // Start the game and print out the gameboard
     printf("Game Started\n");
     print_board(game_board);
 
-    printf("Enter command: ");
-    // TODO (Stage 1.3): Create a command loop, to read and execute commands!
+    // Enter command loop
+    char command;
+    int row, start_col, end_col;
+
+    while (1) {
+        printf("Enter command: ");
+        int result = scanf(" %c", &command);
+        
+        // Exit if CTRL+C or input is invalid
+        if (result == EOF || command == 'q') break;
+        
+        // Handle log command
+        if (command == 'l') {
+            scanf("%d %d %d", &row, &start_col, &end_col);
+
+            // Ensure the row and columns are within bounds and check if row has no turtles
+            if (row > 0 && row < SIZE - 1 && start_col < SIZE && end_col >= 0) {
+                int has_turtle = FALSE;
+                for (int col = 0; col < SIZE; col++) {
+                    if (game_board[row][col].type == TURTLE) {
+                        has_turtle = TRUE;
+                        break;
+                    }
+                }
+
+                // Add logs only if row has no turtles
+                if (!has_turtle) {
+                    for (int col = (start_col < 0 ? 0 : start_col); col <= (end_col >= SIZE ? SIZE - 1 : end_col); col++) {
+                        if (game_board[row][col].type == WATER) {
+                            game_board[row][col].type = LOG;
+                        }
+                    }
```

Comment:
this part can be simpler as well, since we already checked that the `startCol` and `endCol` must be within the bound, and no need to check if water or not, since we know that it's not a turtle.

```suggestion
                    for (int col = startCol; col <= endCol; col++) {
                        game_board[row][col].type = LOG;
                    }
```